elementary/monitor/data_monitoring/data_monitoring_report.py

```diff
@@ -0,0 +1,339 @@
+import json
+import os
+import os.path
+import re
+import webbrowser
+from typing import Dict, List, Optional, Tuple
+
+import pkg_resources
+
+from elementary.clients.gcs.client import GCSClient
+from elementary.clients.s3.client import S3Client
+from elementary.config.config import Config
+from elementary.monitor.api.filters.filters import FiltersAPI
+from elementary.monitor.api.lineage.lineage import LineageAPI
+from elementary.monitor.api.lineage.schema import LineageSchema
+from elementary.monitor.api.models.models import ModelsAPI
+from elementary.monitor.api.models.schema import (
+    ModelRunsSchema,
+    NormalizedExposureSchema,
+    NormalizedModelSchema,
+    NormalizedSourceSchema,
+)
+from elementary.monitor.api.sidebar.schema import SidebarsSchema
+from elementary.monitor.api.sidebar.sidebar import SidebarAPI
+from elementary.monitor.api.tests.schema import TotalsSchema
+from elementary.monitor.api.tests.tests import TestsAPI
+from elementary.monitor.data_monitoring.data_monitoring import DataMonitoring
+from elementary.monitor.data_monitoring.schema import DataMonitoringFilter
+from elementary.tracking.anonymous_tracking import AnonymousTracking
+from elementary.utils.log import get_logger
+from elementary.utils.time import get_now_utc_iso_format
+
+logger = get_logger(__name__)
+
+YAML_FILE_EXTENSION = ".yml"
+SQL_FILE_EXTENSION = ".sql"
+
+
+class DataMonitoringReport(DataMonitoring):
```

Comment:
Did a bit of an order in the mess we got here.
Added support for invocation filter as well